src/rest_vol_object.c

```diff
@@ -843,6 +861,33 @@ RV_object_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_object_s
                             strncpy(iter_object->u.file.filepath_name, loc_obj->u.file.filepath_name,
                                     strlen(loc_obj->u.file.filepath_name) + 1);
 
+                            if ((iter_object->u.file.server_info.username =
```

Comment:
Won't `iter_object` here already be inheriting the file/domain from `loc_obj`? If so, it seems like we'd be overwriting and leaking memory by allocating the file's `server_info` fields here

Reply:
`iter_object` has the `server_info` struct's pointers copied into it, pointing at the allocated memory of the original file object. If we didn't allocate more memory here, the server information of the original would be freed when this copy is closed.